src/operator/timeout.ts

```diff
@@ -45,6 +46,8 @@ class TimeoutOperator<T> implements Operator<T, T> {
 class TimeoutSubscriber<T> extends Subscriber<T> {
   private index: number = 0;
   private _previousIndex: number = 0;
+  private action: Subscription = null;
+
   get previousIndex(): number {
```

Comment:
not sure, but `hasCompleted` indeed seems extraneous. I couldn't figure out why there is an index (aka count) check anyway? Why would it ever not be equal?